controllers/Backend.php

```diff
@@ -1,104 +1,352 @@
+                    $errors['empty'] = "Veuillez remplier tous les champs";
+                }
+
+                if($email != $email_again){
+                    $errors['different'] = "Les adresses email ne correspondent pas";
+                }
+
+                if($manager_user->email_taken($email)){
+                    $errors['taken'] = "L'adresse email est déjà assignée à un modérateur";
+                }
+
+                if(empty($errors)){
+                    $manager_user->add_modo($name,$email,$role,$token);
+                    header("Location:/gestion-des-admins-et-modos");
+                }
+            }
+            else {
+                $errors = [];
+            }
+            
+        if(!isset($_SESSION['admin'])){
+            header("location: /articles-a-la-une");
+        }
         
-        $page="backend/settings";
-        $topbar="backend/topbar";
-        require "views/layout.php";
+        $myView = new View('settings', 'backend');
+        $myView->render(array('modos' => $modos, 'errors' => $errors));
     }
 
     public function list()
     {
-        $model_article = new Article();
-        $model_user = new User();
-
-        $page="backend/list";
-        $topbar="backend/topbar";
-        require "views/layout.php";
-    }
+        $manager_article = new ArticleManager();
+        $posts = $manager_article->get_posts();
 
-    public function logout()
-    {
-        $model_user = new User();
+        if(!isset($_SESSION['admin'])){
+            header("location: /articles-a-la-une");
+        }
         
-        $page="backend/logout";
-        $topbar="backend/topbar";
-        require "views/layout.php";
+        $myView = new View('list', 'backend');
+        $myView->render(array('posts' => $posts));
     }
 
     public function password()
     {
-        $model_user = new User();
+        $manager_user = new UserManager();
+        $errors = $manager_user->password_verification();
+   
+         if(filter_has_var(INPUT_POST, 'submit')){
+            $password = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'password')));
+            $password_again = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'password_again')));
+
+            $errors = [];
+            
+            if(empty($password) || empty($password_again)){
+                $errors['empty'] = "Tous les champs n'ont pas été remplis";
+            }
+
+            if($password != $password_again){
+                $errors['different'] = "Les mots de passe sont différents";
+            }
+
+            if (preg_match('#^(?=.*[a-z])(?=.*[A-Z]).{8,}$#', $password)) {
+            }
+            else{
+                $errors['non conforme'] = 'Votre mot de passe doit contenir des minuscules et des majuscules et posséder une longueur de 8 caractères au minimum';
+            }	
+                    
+            if(empty($errors)){
+                Usermanager::update_password($password);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid using static access to class 'Usermanager' in method 'password'.](https://app.codacy.com/manual/JeromeLacquemant/projet-5/pullRequest?prid=5379676)